src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/service/CertificateService.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.tractusx.bpdmcertificatemanagement.service
+
+import mu.KotlinLogging
+import org.eclipse.tractusx.bpdmcertificatemanagement.dto.request.CertificateDocumentRequestDto
+import org.eclipse.tractusx.bpdmcertificatemanagement.dto.response.CertificateDocumentResponseDto
+import org.eclipse.tractusx.bpdmcertificatemanagement.entity.CertificateTypeDB
+import org.eclipse.tractusx.bpdmcertificatemanagement.exception.CertificateTypeNotExists
+import org.eclipse.tractusx.bpdmcertificatemanagement.repository.CertificateRepository
+import org.eclipse.tractusx.bpdmcertificatemanagement.repository.CertificateTypeRepository
+import org.springframework.stereotype.Service
+
+@Service
+class CertificateService(
+    val certificateRepository: CertificateRepository,
+    val certificateTypeRepository: CertificateTypeRepository,
+    val certificateMapping: CertificateMapping
+) {
+
+    private val logger = KotlinLogging.logger { }
+
+    fun createCertificate(certificateDocumentRequestDto: CertificateDocumentRequestDto): CertificateDocumentResponseDto? {
+        logger.debug { "Executing createCertificate() with parameters $certificateDocumentRequestDto" }
+
+        val certificateType =
+            certificateTypeRepository.findByCertificateType(certificateDocumentRequestDto.type.certificateType)
```

Comment:
You just use certificateType as key to identify the certificate type. But in the request the user has to give the certificateType key and the version. If only the type is enough why should the user need to specify also the version?

Reply:
There is an argument to made on this, i believe certificateType is fine yeah i can make it with version also but while creating checking and adding to certificate it is not necessary as while adding certificateType in DB version cannot be null. Please let me know if you have other opinion on same.

Reply:
In the certificate request I would just use the certificate type key value to reference the type. When creating or updating a certificate type I would use the existing Certificate request DTO and the respective metadata endpoints.

In the certficate request DTO you just want a reference to a certificate value so the one type value is enough. I think it is confusing for the user to also be able to put in the certificate version and this attribute is completely ignored.

Reply:
Yeah, Its my bad!!
I checked requirement document again where it is mentioned both certificateType and certificateVersion has to be mentioned in type. 
I am adding both keys while identifying correct certificate type.